src/python/zquantum/qcbm/target_thermal_states.py

```diff
@@ -0,0 +1,235 @@
+import bisect
+import typing
+import numpy as np
+from zquantum.core.bitstring_distribution import BitstringDistribution
+from zquantum.core.utils import dec2bin, bin2dec, convert_tuples_to_bitstrings
+
+
+def int2ising(number: int, length: int) -> typing.List[int]:
+    """Converts an integer into a +/-1s bitstring (also called Ising bitstring).
+    Args:
+        number: positive number to be converted into its corresponding Ising bitstring representation
+        length: length of the Ising bitstring (i.e. positive number of spins in the Ising system)
+    Returns:
+        Ising bitstring representation (1D array of +/-1)."""
+
+    binary_bitstring = dec2bin(number, length)
+    ising_bitstring = [bit * 2 - 1 for bit in binary_bitstring]
+    return ising_bitstring
+
+
+def ising2int(ising_bitstring: typing.List[int]) -> int:
+    """Converts a +/-1s bitstring (also called Ising bitstring) into an integer.
+    Args:
+        ising_bitstring: 1D array of +/-1.
+    Returns:
+        Integer number representation of the Ising bitstring."""
+
+    binary_bitstring = [int((bit + 1) / 2) for bit in ising_bitstring]
+    number = bin2dec(binary_bitstring)
+    return number
+
+
+def get_random_hamiltonian_parameters(n_spins: int) -> typing.Tuple:
```

Comment:
This function name could also be more descriptive as it's not obvious what hamiltonian parameters are being randomly generated.

Reply:
Also, the types within the output tuple should be added to the function header.

Reply:
Lastly, if this function is only used in a certain place, it might be better to make it private.

Reply:
I really don't know of a better name for this - even though I know this one is quite generic. 
Types = done. Private = done.

Reply:
Change it to get_random_ising_hamiltonian_parameters and add details to docstring

Reply:
Input the name of the hamiltonian model - didn't write out the formula.

Reply:
Made a small correction (TFIM -> IM with non-zero external fields)